addons/mail/static/src/js/owl/component/drop_zone.js

```diff
@@ -0,0 +1,83 @@
+odoo.define('mail.component.DropZone', function () {
+'use strict';
+
+class DropZone extends owl.store.ConnectedComponent {
+    /**
+     * @param {...any} args
+     */
+    constructor(...args) {
+        super(...args);
+        this.id = _.uniqueId('o_dropZone_');
+        this.template = 'mail.component.DropZone';
+        this.state = { draggingInside: false };
+        this._globalDragOverListener = ev => this._onDragOver(ev);
+        this._globalDropListener = ev => this._onDrop(ev);
+    }
+
+    mounted() {
+        document.addEventListener('dragover', this._globalDragOverListener);
+        document.addEventListener('drop', this._globalDropListener);
+    }
+
+    /**
+     * Making sure that dragging content is external files.
+     * Ignoring other content draging like text.
+     *
+     * @private
+     * @param {DataTransfer} dataTransfer
+     * @returns {boolean}
+     */
+    _isDragSourceExternalFile(dataTransfer) {
+        const DragDataType = dataTransfer.types;
```

Comment:
no capital first char for non-class objects

Reply:
copy pasted from actual discuss module, thought it was done on purpose ; changed it :)